__tests__/e2e_timer_tests.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-promise-executor-return */
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-return-assign */
+/* eslint-disable no-undef */
+
+/**
+ * @jest-environment jsdom
+ */
+
+/**
+ * File: e2e_timer_tests.js
+ * Author: Manshi Yang
+ * Description: Include some e2e tests for timer
+ * Note: Please run with the command 'npm run test --detectOpenHandles'
+ */
+const puppeteer = require('puppeteer');
+
+// TODO: replace with the correct web link
+const URL = 'http://127.0.0.1:5500/source/index.html';
```

Comment:
This needs to direct to the Netlify deployment, not liveserver. Otherwise this will always fail since there is no liveserver environment in the GitHub actions server. It also breaks on local machines without liveserver running.

Reply:
Fixed in GitHub editor.